plans/dht/test/set-up.go

```diff
@@ -0,0 +1,116 @@
+package test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ipfs/testground/sdk/runtime"
+	"github.com/ipfs/testground/sdk/sync"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	kaddht "github.com/libp2p/go-libp2p-kad-dht"
+)
+
+// SetUp sets up the elements necessary for the test cases
+func SetUp(ctx context.Context, runenv *runtime.RunEnv, timeout time.Duration, randomWalk bool, bucketSize int, autoRefresh bool) (host.Host, *kaddht.IpfsDHT, []peer.AddrInfo, error) {
+	/// --- Set up
+
+	node, dht, err := CreateDhtNode(ctx, runenv, bucketSize, autoRefresh)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	watcher, writer := sync.MustWatcherWriter(runenv)
+	defer watcher.Close()
+	defer writer.Close()
+
+	/// --- Tear down
+
+	defer func() {
```

Comment:
Won't this end the entire _test_ when we exit this function?

Reply:
Still needs to wait for the barrier to signal that it is complete.

Reply:
Previously, this barrier would happen after the _test_ completes. Now, it happens after _setup_ completes.

Reply:
You are right. Fixing it

Reply:
fixed